src/main/java/seedu/address/model/person/Person.java

```diff
@@ -36,6 +39,22 @@ public Person(Name name, Phone phone, Email email, Address address, Remark remar
         this.address = address;
         this.tags.addAll(tags);
         this.remark = remark;
+        this.status = new LeadStatus(LeadStatusName.UNCONTACTED.getLabel());
+    }
+
+    /**
+     * Constructor for a person with a given LeadStatus.
+     */
+    public Person(Name name, Phone phone, Email email, Address address, Remark remark, Set<Tag> tags,
+                     LeadStatus status) {
+        requireAllNonNull(name, phone, email, address, tags, remark);
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.address = address;
+        this.tags.addAll(tags);
+        this.remark = remark;
+        this.status = status;
     }
```

Comment:
```suggestion
    public Person(Name name, Gender gender, Phone phone, Email email, Company company, Industry industry, Occupation occupation, JobTitle jobTitle, Address address, Remark remark, Set<Tag> tags, LeadStatus status) {
        requireAllNonNull(name, gender, phone, email, company, industry, occupation, jobTitle, address, tags, remark, status);
        this.name = name;
        this.gender = gender;
        this.phone = phone;
        this.email = email;
        this.address = address;
        this.company = company;
        this.industry = industry;
        this.occupation = occupation;
        this.jobTitle = jobTitle;
        this.tags.addAll(tags);
        this.remark = remark;
        this.status = status;
    }
```

Would need to include these new parameters afterwards. 
Also, not sure if intentional but might have missed out adding status to requireAllNonNull(...) method.

The other Person constructor should not need any changing as I assume merging both our branches will rectify any differences.

Reply:
Thanks for pointing this out – you're right that the non-null check should include the new attributes.

I have since added it with the bug fixes for Person constructor.